### PR TITLE
feat(dedibox): add waiter support for Service and Server

### PR DESCRIPTION
**Pull request by Laure-di**

### CODE REVIEW

File: api/dedibox/v1/dedibox_utils.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import (
+	"github.com/scaleway/scaleway-sdk-go/internal/async"
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"time"
+)
+
+const (
+	defaultRetryInterval = time.Second * 15
+	defaultTimeout       = time.Second * 30
```
Comment (yfodil) [inline api/dedibox/v1/dedibox_utils.go:12]: 30s isnt a bit too low for a Dedibox server creation ?

Comment (Laure-di) [reply api/dedibox/v1/dedibox_utils.go:12]: Oups went too fast means minutes

File: api/dedibox/v1/dedibox_utils.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+const (
+	defaultRetryInterval = time.Second * 15
+	defaultTimeout       = time.Minute * 30
+)
+
+type WaitForServiceRequest struct {
+	ServiceID     uint64
+	Zone          scw.Zone
+	Status        ServiceProvisioningStatus
```
Comment (yfodil) [inline api/dedibox/v1/dedibox_utils.go:18]: is this field necessary for this request?

File: api/dedibox/v1/dedibox_utils.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		return nil, errors.Wrap(err, "waiting for service failed")
+	}
+	return service.(*Service), nil
+}
+
+type WaitForServerRequest struct {
+	ServerID      uint64
+	Zone          scw.Zone
+	Status        ServerStatus
```
Comment (yfodil) [inline api/dedibox/v1/dedibox_utils.go:61]: is this field necessary for this request?

File: api/dedibox/v1/dedibox_utils.go (hunk 2 above)
Comment (Laure-di) [reply api/dedibox/v1/dedibox_utils.go]: Fix :)

File: api/dedibox/v1/dedibox_utils.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	timeout := defaultTimeout
+	if req.Timeout != nil {
+		timeout = *req.Timeout
+	}
+	retryInterval := defaultRetryInterval
+	if req.RetryInterval != nil {
+		retryInterval = *req.RetryInterval
+	}
+	terminalStatus := map[ServerStatus]struct{}{
+		ServerStatusReady:  {},
```
Comment (yfodil) [inline api/dedibox/v1/dedibox_utils.go:74]: Some terminal statuses are missing: stopped, rescue and busy

Comment (Laure-di) [reply api/dedibox/v1/dedibox_utils.go:74]: there are not terminal status

**Review by yfodil — APPROVED** [review]

**Pull request merged**